framework/helpers/BaseFileHelper.php

```diff
@@ -253,6 +253,9 @@ protected static function loadMimeTypes($magicFile)
      */
     public static function copyDirectory($src, $dst, $options = [])
     {
+        if ($src === $dst || strpos($dst, $src) !== false){
```

Comment:
strpos should only match if its result is 0, e.g. copying from `/data` to `/backup/data` should work.
Would be good to also add tests to https://github.com/yiisoft/yii2/blob/master/tests/framework/helpers/FileHelperTest.php